From 089722c8c0bc2c99fa3692e30e936e2d2438b0f6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Wed, 28 Aug 2024 21:28:52 +0200
Subject: [PATCH] downloader: Don't consider derives-from field for logos

The 'derives-from' field can assign different distributions together, which
causes that wrong logo is displayed for an unofficial fork. E.g. PureOS is
displayed with Debian logo, which shouldn't happen. Consider only the
'upgrades' field, which is always official.
---
 src/downloader.vala | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/src/downloader.vala b/src/downloader.vala
index 02504f705..335bf00a8 100644
--- a/src/downloader.vala
+++ b/src/downloader.vala
@@ -40,9 +40,7 @@ public static string fetch_os_logo_url (Osinfo.Os os) {
             return os.logo;
 
         string logo_url = null;
-        var derived = os.get_related (Osinfo.ProductRelationship.DERIVES_FROM);
-        if (derived.get_length () == 0)
-            derived = os.get_related (Osinfo.ProductRelationship.UPGRADES);
+        var derived = os.get_related (Osinfo.ProductRelationship.UPGRADES);
 
         while (derived.get_length () > 0 && logo_url == null) {
             // FIXME: Does Osinfo allows deriving from multiple OSs?
@@ -51,7 +49,7 @@ public static string fetch_os_logo_url (Osinfo.Os os) {
             if (parent.logo != null)
                 logo_url = parent.logo;
             else
-                derived = parent.get_related (Osinfo.ProductRelationship.DERIVES_FROM);
+                derived = parent.get_related (Osinfo.ProductRelationship.UPGRADES);
         }
 
         return logo_url;
-- 
GitLab

